packages/oui-file/src/file.controller.js

```diff
@@ -0,0 +1,204 @@
+import { addBooleanParameter, addDefaultParameter } from "@ovh-ui/common/component-utils";
+import find from "lodash/find";
+import isEmpty from "lodash/isEmpty";
+import remove from "lodash/remove";
+
+export default class {
+    constructor ($attrs, $compile, $element, $filter, $scope, $timeout, $window, ouiFileConfiguration) {
+        "ngInject";
+
+        this.$attrs = $attrs;
+        this.$compile = $compile;
+        this.$element = $element;
+        this.$filter = $filter;
+        this.$scope = $scope;
+        this.$timeout = $timeout;
+        this.$window = $window;
+
+        this.translations = ouiFileConfiguration.translations;
+        this.units = this.$filter("orderBy")(ouiFileConfiguration.units, "-size");
+    }
+
+    checkFileValidity (file) {
+        if (file) {
+            file.errors = {};
+
+            // Check maxsize
+            if (this.maxsize && file.size > this.maxsize) {
+                file.errors.maxsize = true;
+            }
+
+            // Set form validation
+            if (this.form && this.form[this.name]) {
+                this.form[this.name].$setValidity("maxsize", !file.errors.maxsize);
+                this.form[this.name].$setDirty();
+            }
+
+            // Clean errors
+            if (isEmpty(file.errors)) {
+                delete file.errors;
+            }
+        }
+
+        return file;
+    }
+
+    loadFilePreview (file) {
+        // Load preview only if image
+        if (this.preview && !file.errors && file.type && file.type.search(/^image\//) !== -1) {
+            file.loading = true;
+
+            file.reader = new this.$window.FileReader();
+            file.reader.readAsDataURL(file);
+            file.reader.onload = () => {
+                file.preview = `url("${file.reader.result}")`;
+
+                // Clean file object
+                delete file.loading;
+                delete file.reader;
+
+                this.$scope.$apply();
+            };
+        }
+
+        return file;
+    }
+
+    addFile (file) {
+        this.model = [this.checkFileValidity(file)];
```

Comment:
Won't the `model` only contain last file ? Also, you still add the file even though it's invalid or empty ?

Reply:
Just realized you have a `addFiles` method. I would have had only one method that can handle the case where their are one or more than one file